### DISCUSSION TITLE
Agent Automation fails when using more than one required attribute

### DISCUSSION THREAD

**Discussion by stjohnson44** [twilio-professional-services/flex-project-template#680] [Q&A] [answered]
We have recently been exploring using the agent automation feature after using our own implementation for some time, simply to cut back on the number of listeners being added at different points. When investigating, I found that, due to how `getMatchingTaskConfiguration()` is written, unless the matching attribute is the final one checked, the check will always return false.

For instance, I could set the configuration to be the following:

```
[
  {
    "auto_accept": true,
    "auto_select": true,
    "auto_wrapup": false,
    "channel": "voice",
    "default_outcome": "Automatically completed",
    "required_attributes": [
      {
        "key": "direction",
        "value": "inbound"
      },
      {
        "key": "taskType",
        "value": "predictive-dial"
      },
      {
        "key": "taskType",
        "value": "message-dial"
      }
    ],
    "wrapup_time": 30000
  }
]
```

If I then make an inbound call, while I expect the call to be automatically accepted, it is not, because it continues checking the remaining attributes, and when that check fails, the boolean is set back to false.  The same is true when checking worker attributes. I would recommend doing something similar to the following when checking the required attributes and the required worker attributes:

```
for (let i = 0; i < config.required_attributes?.length; i++) {
  if (attributes[config.required_attributes[i].key] === config.required_attributes[i].value) {
    matched_config = true;
    break;
  } else {
    matched_config = false;
  }
}
```

This allows us to break  the loops when a matching condition is found, which using .forEach does not provide.

**Comment by dremin** [comment, marked as answer]
Hi @stjohnson44! I think there's a slight misunderstanding here--the intent with the logic as it stands is that **all** of the attributes listed must match (i.e. it's an AND condition, not an OR). So for your use case, you'd create two different configs--one config with direction=inbound and taskType=predictive-dial, and another config with direction=inbound and taskType=message-dial.

**Reply by stjohnson44** [reply]
Ahh, thanks for the clarification. That makes much more sense now!
